[MIG] datetime formatter: Migration to 15.0

tbanevicius opened the pull request.

Commit:
Add module datetime_formatter.

This module allows the developer to format any date/time combination in
the most comfortable way for the user, just with a quick call to

oca-transbot committed:
OCA Transbot updated translations from Transifex

cubells committed:
datatime_formatter migration to 9.0

yajo committed:
[MIG][10.0][datetime_formatter] Migration

Special highlights:

- Fixed some license headers.
- Adhered to normal addon structure.
- Removed `exceptions.py` and use `UserError` instead.
- Allow submodules to import some constants from main namespace.

The rest is basically a standard migration.

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[MIG] datetime_formatter: Migration to 11.0

oca-travis committed:
[UPD] Update datetime_formatter.pot

ernestotejeda committed:
[MIG] datetime_formatter: Migration to 12.0

oca-travis committed:
[UPD] Update datetime_formatter.pot

OCA-git-bot committed:
[UPD] README.rst

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (2 of 2 strings)

Translation: server-tools-12.0/server-tools-12.0-datetime_formatter
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-datetime_formatter/pt/

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (2 of 2 strings)

Translation: server-tools-12.0/server-tools-12.0-datetime_formatter
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-datetime_formatter/zh_CN/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (2 of 2 strings)

Translation: server-tools-12.0/server-tools-12.0-datetime_formatter
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-datetime_formatter/hr/

victoralmau committed:
[IMP] datetime_formatter: black, isort, prettier

victoralmau committed:
[MIG] datetime_formatter: Migration to 13.0

oca-travis committed:
[UPD] Update datetime_formatter.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-13.0/server-tools-13.0-datetime_formatter
Translate-URL: https://translation.odoo-community.org/projects/server-tools-13-0/server-tools-13-0-datetime_formatter/

dsolanki-initos committed:
[MIG] Migrate module datetime_formatter to v14.

oca-travis committed:
[UPD] Update datetime_formatter.pot

OCA-git-bot committed:
[UPD] README.rst

TautvydasBanevicius committed:
[IMP] datetime_formatter: black, isort, prettier, pylint, flake8.

Modified exception on method best_match according flake8 warning.
Modified test method test_user with_context(**) according pyling warning.

TautvydasBanevicius committed:
[MIG] datetime_formatter: Bumped version to 15.0

tbanevicius commented:
Can someone review this merge request? 
It seems CI/runbot fails without any eligible reason.

tbanevicius commented:
/ocabot migration auditlog

tbanevicius commented:
@dreispt @guewen @pedrobaeza Is there any chance someone could review this MR?

pedrobaeza commented:
Have you signed CLA?

tbanevicius commented:
@pedrobaeza I did signed CLA. Sent signed **ICLA** document to **cla@odoo-community.org** on February 28.

pedrobaeza commented:
Now you can review other PRs and ask in exchange that they review yours.

aisopuro commented:
The actual diff of the migration is here: https://github.com/OCA/server-tools/compare/723ac37584fa0b136edc546c119b83d7c95c0160...tbanevicius:15.0-mig-datetime-formatter

aisopuro commented on datetime_formatter/tests/test_best_matcher.py:
```diff
@@ -50,7 +50,9 @@ def test_user(self):
             # Lang not found in context
-            self.assertEqual(self.rl.with_context(dict()).best_match().code, lang.code)
+            self.assertEqual(
+                self.rl.with_context(**dict()).best_match().code, lang.code
```
`pylint` seems to be mistaken here, this does not seem like a safe change.

The old call in version 14 used an empty dict, which means `with_context` empties the context, removing any `lang` key if it was present.

However this new version unpacks an empty dictionary, which will not override anything. In particular in this test, if `lang` is in the context, this will _not_ clear it, thus making it uncertain whether this test is actually testing the correct functionality.
To placate `pylint`, consider changing it to perhaps use `{}` instead of `dict()`. If that doesn't work, you may be able to [explicitly ignore it](https://stackoverflow.com/a/29562085).

tbanevicius replied:
{} did the trick. 
Now pylint has some issues with readme file, though README.rst, readme folder with CONTRIBUTORS.rst, DESCRIPTION.rst, USAGE.rst is present in the module..

aisopuro reviewed (changes requested):
One small linting conflict.

tbanevicius commented:
Hi @Yajo @victoralmau  any idea why pylint catches error on README.rst? Even though it exists there..

yajo commented:
No idea, it's weird indeed 😕 

Any idea @sbidoul @moylop260?

moylop260 commented:
> No idea, it's weird indeed 😕
> 
> Any idea @sbidoul @moylop260?

It looks like `sample_module` was identified as a real module in pylint + pre-commit

Since that pre-commit run for each individual path instead of run over the root path only.

I mean,

old: `pylint module/`
new (pre-commit): `pylint module_auto_update/tests/sample_module/models/stuff.py; pylint module_auto_update/tests/sample_module/models/otherfiles`

sbidoul commented:
@moylop260 your explanation makes sense. Not sure why the error did not trigger in 14.0 though.
Anyway, the easy fix should be to add a README.rst in that `sample_module`.

tbanevicius committed:
[IMP] module_auto_update: Improved README.rst

sample_module was identified as a real module in pylint + pre-commit.
Updated README.rst

tbanevicius commented:
@moylop260  @Yajo  @sbidoul  thanks! Seems now all checks have passed.

tbanevicius commented:
@Yajo could you review this MR?

yajo commented:
/ocabot merge nobump
